File: Array/liuchang/day37/day37.java
public class day37 {


    public static String changeString(String str){
        char[] chars = str.toCharArray();
        int count = chars.length;
        char newString[] = new char[count];
        for (int i = count - 1; i >= 0; i--) {
            for (int j = 0; j < count ; ++j) {
                if (newString[j]=='\0'){
                    newString[j] = chars[i];
                    break;
                }
            }
        }
        return String.valueOf(newString);

    }


    public static void main(String[] args) {
        System.out.println(changeString("abcdef"));
    }

}

File: Array/liuchang/day37/day37.kt
package net.study.kotlintest

/**
 * @author       刘畅
 * @createdate   2020/7/7
 * @describe
 */
class day37 {


    fun changeString(str: String): String {
        var chars: CharArray = str.toCharArray();
        var count = chars.size;
        var newString = CharArray(count);
        for (i in count - 1 downTo 0) {
            for (j in 0 until count) {
                if (newString[j] == '\u0000') {
                    newString[j] = chars[i];
                    break;
                }
            }
        }
        return String(newString);

    }

    fun main(args:Array<String>){
        System.out.println("====="+changeString("abcdef"))
    }
}